js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +34,86 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      clicked: 0,
     }
   }
+  /**
+   * Change the clicked and courseId state, whenever a course link is clicked.
+   */
+  clickRelated = s => {
+    let copyState = this.state
```

Comment:
In general, you shouldn't update state like this. Instead, pass into `setState` a new object that has just the state values you want to update, e.g. `this.setState({courseId: s, clicked: 1})`.

Reply:
Additionally: it's *very* important to note that assigning `let copyState = this.state` does **not** create a copy of the object, but rather creates an alias. The name copy, here and further below, is very misleading and should be avoided.